### DISCUSSION TITLE
Use Property wrappers on the @ObservableState properties

### DISCUSSION THREAD

**Discussion by hadiidbouk** [pointfreeco/swift-composable-architecture#2938] [Q&A] [answered]
I have a `Defaults` property wrapper that saves and reads from user defaults.

I used to mark any property in my states with it, like the following:

```swift
@Reducer
struct MyFeature {
    struct State {
        @Defaults(key: "isDebugModeEnabled", defaultValue: false)
	var isDebugModeEnabled: Bool
    }
}
```

But now with the `@ObservableState` macro, these properties are converted to computed ones, and I cannot use my property wrapper anymore.

```swift
@Reducer
struct MyFeature {
    @ObservableState
    struct State {
        @Defaults(key: "isDebugModeEnabled", defaultValue: false)  <-- Property wrapper cannot be applied to a computed property
	var isDebugModeEnabled: Bool
    }
}
```

After seeing this, won't the user expect when he writes the above that the `isDebugModeEnabled` is a stored property?
I think this part could be confusing.

What's the recommended approach here?

**Comment by mbrandonw** [comment, marked as answer]
Hi @hadiidbouk, this is just an issue with Swift macros in general, in particular ones that turn a stored property into a stored+computed property duo. This affects Apple's own `@Observable` and `@Model` macros too.

There really is no universal, good solution to this problem. All I can say is that we are currently running a [shared state beta](https://github.com/pointfreeco/swift-composable-architecture/discussions/2857) right now that solves the particular problem you are having, that of backing a field with user defaults. You will be able to use it like so:

```swift
@Shared(.appStorage("isDebugModeEnabled")) var isDebugModeEnabled: Bool
```

So, you can either give the beta a spin now, or wait until the tools officially ship sometime in the near future.
